sharding/indexHashedNodesCoordinator.go

```diff
@@ -882,3 +885,36 @@ func createValidatorInfoFromBody(
 
 	return allValidatorInfo, nil
 }
+
+func toOrderedSlice(validatorMap map[uint32][]Validator, numShardsPlusMeta uint32) [][]Validator {
+	slicedLists := make([][]Validator, numShardsPlusMeta)
+	for shardId, list := range validatorMap {
+		if shardId < numShardsPlusMeta-1 {
+			slicedLists[shardId] = list
```

Comment:
This will break in panic if len(map[uint32][]Validator) > numShardsPlusMeta . Not saying that it is possible as the code is written (the method is unexported and the parameters might be carefully crafted) but the code just become unnecessarily fragile. 
Strongly suggest to do this in 3 easy steps:
1. collect keys (shard ids) into a slice of uint32
2. sort that slice
3. iterate the keys

Will work with every map, in any configuration.

A better, cleaner solution is to have a DTO that will have 2 fields: the validator list and the shardID, then you could do it in 2 easy steps:
1. iterate the map, contruct the DTO for each shard, append it to the slice
2. sort the resulting DTO slice by the shard ID field

The advantage of the latest is that the code will become easier to read, not to mention the fact that now we have to keep in mind that the last item is the metachain's ID. Also, there you will not need the numShardsPlusMeta variable injected just about everywhere.

Reply:
undone. used sorted keys on map